source/transfers/contracts/TransferHandlerRegistry.sol

```diff
@@ -0,0 +1,55 @@
+pragma solidity 0.5.12;
+
+import "./interfaces/ITransferHandler.sol";
+
+/**
+  * @title TransferHandlerRegistry: holds registry of contract to
+  * facilitate token transfers
+  */
+contract TransferHandlerRegistry {
+
+  event AddTransferHandler(
+    bytes4 kind,
+    address contractAddress
+  );
+
+  event RemoveTransferHandler(
+    bytes4 kind,
+    address contractAddress
+  );
+
+  // Mapping of bytes4 to contract interface type
+  mapping (bytes4 => ITransferHandler) private transferHandlerMapping;
+
+  /**
+  * @notice Adds handler to mapping
+  * @param _kind bytes4
+  * @param _transferHandler ITransferHandler
+  */
+  function addTransferHandler(bytes4 _kind, ITransferHandler _transferHandler)
+    external {
+      transferHandlerMapping[_kind] = _transferHandler;
```

Comment:
There's no checks for whether the entry is already taken. This means if we add a handler for ERC20 tokens, someone could override that entry with their own handler. Is that intended behaviour?

Reply:
removed.